src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeValueType.php

```diff
@@ -78,7 +78,7 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
         ;
 
         $builder->get('localeCode')->addModelTransformer(
-            new ReversedTransformer(new ResourceToIdentifierTransformer($this->localeRepository, 'code')),
+            new ReversedTransformer(new ResourceToIdentifierCacheableTransformer($this->localeRepository, 'code')),
```

Comment:
I'm wondering whether we shouldn't add a new `LocaleToCodeTransformer` in the locale bundle independently from resource bundle.
In the case of locales, currencies, countries (and I'm guessing channels as well) transforming from and to code is used quite heavily. We could then leverage symfony's cache mechanism, so in the best-case scenario we'd have 0 queries and the cache would stay between requests.
Additionally, since we'd have one transformer per the aforementioned resource we could make them a normal service instead of recreating the object dynamically on each call.

Reply:
The idea initially sounds good for me, but I'm not sure this is a thing to be done during that PR. It should be rather done through entire application I think.